pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -95,6 +96,14 @@ func (b *AutoscalingGroupModelBuilder) buildLaunchTemplateTask(c *fi.ModelBuilde
 		return nil, err
 	}
 
+	// Launch template needs the User Data to be Base64 encoded.
+	d, err := lc.UserData.AsBytes()
+	if err != nil {
+		return nil, err
+	}
+	b64d := base64.StdEncoding.EncodeToString(d)
+	b64UserDataResource := fi.WrapResource(fi.NewStringResource(b64d))
```

Comment:
Might be nicer to pass the non-base64 encoded version into the task, and then do the base64 encoding in the task itself.  This is because our diff engine should show the diff for the UserData field when it changes, but it's not going to be very readable when base64 encoded :-)